com.unity.render-pipelines.core/ShaderLibrary/HDROutput.hlsl

```diff
@@ -0,0 +1,657 @@
+#ifdef HDR_OUTPUT_SCRGB
+    inputColor = RotateRec709ToRec2020(inputColor);
+#endif
+    return RotateRec2020ToICtCp(inputColor);
+}
+
+float3 RotateLMSToXYZ(float3 LMSInput)
+{
+    static const float3x3 LMSToXYZMat = float3x3(
+        2.07018005669561320f, -1.32645687610302100f, 0.206616006847855170f,
+        0.36498825003265756f, 0.68046736285223520f, -0.045421753075853236f,
+        -0.04959554223893212f, -0.04942116118675749f, 1.187995941732803400f
+        );
+    return mul(LMSToXYZMat, LMSInput);
+}
+
+float3 RotateXYZToRec2020(float3 XYZ)
+{
+    static const float3x3 XYZToRec2020Mat = float3x3(
+        1.71235168f, -0.35487896f, -0.25034135f,
+        -0.66728621f, 1.61794055f,  0.01495380f,
+        0.01763985f, -0.04277060f,  0.94210320f
+    );
+
+    return mul(XYZToRec2020Mat, XYZ);
+}
+
+float3 RotateXYZToRec709(float3 XYZ)
+{
+    return mul(XYZ_2_REC709_MAT, XYZ);
+}
+
+float3 RotateRec709ToXYZ(float3 rgb)
+{
+    static const float3x3 Rec709ToXYZMat = float3x3(
+        0.412391f, 0.357584f, 0.180481,
+        0.212639, 0.715169, 0.0721923,
+        0.0193308, 0.119195, 0.950532
+        );
+    return mul(Rec709ToXYZMat, rgb);
+}
+
+float3 RotateRec2020ToXYZ(float3 rgb)
+{
+    static const float3x3 Rec2020ToXYZMat = float3x3(
+        0.638574, 0.144617, 0.167265,
+        0.263367, 0.677998, 0.0586353,
+        0.0f, 0.0280727, 1.06099
+        );
+
+    return mul(Rec2020ToXYZMat, rgb);
+}
+
+float3 RotateICtCpToPQLMS(float3 ICtCp)
+{
+    static const float3x3 ICtCpToPQLMSMat = float3x3(
+        1.0f, 0.0086051456939815f, 0.1110356044754732f,
+        1.0f, -0.0086051456939815f, -0.1110356044754732f,
+        1.0f, 0.5600488595626390f, -0.3206374702321221f
+    );
+
+    return mul(ICtCpToPQLMSMat, ICtCp);
+}
+
+float3 RotateICtCpToXYZ(float3 ICtCp)
+{
+    float3 PQLMS = RotateICtCpToPQLMS(ICtCp);
+    float3 LMS = PQToLinear(PQLMS, MAX_PQ_VALUE);
+    return RotateLMSToXYZ(LMS);
+}
+
+float3 RotateICtCpToRec2020(float3 ICtCp)
+{
+    return RotateXYZToRec2020(RotateICtCpToXYZ(ICtCp));
+}
+
+float3 RotateICtCpToRec709(float3 ICtCp)
+{
+    return RotateXYZToRec709(RotateICtCpToXYZ(ICtCp));
+}
+
+float3 RotateICtCpToOutputSpace(float3 ICtCp)
+{
+#ifdef HDR_OUTPUT_SCRGB
+    return RotateICtCpToRec709(ICtCp);
+#elif defined(HDR_OUTPUT_REC2020)
+    return RotateICtCpToRec2020(ICtCp);
+#else
+    return RotateICtCpToRec2020(ICtCp);
+#endif
+
+}
+
+// --------------------------------------------------------------------------------------------
+
+// --------------------------------
+//  OETFs
+// --------------------------------
+// The functions here are OETF, technically for applying the opposite of the PQ curve, we are mapping
+// from linear to PQ space as this is what the display expects.
+// See this desmos for comparisons https://www.desmos.com/calculator/5jdfc4pgtk
+#define PRECISE_PQ 0
+#define ISS_APPROX_PQ 1
+#define GTS_APPROX_PQ 2
+
+#define OETF_CHOICE GTS_APPROX_PQ
```

Comment:
is it worse it? the PRECISE_PQ dont look like so expensive. Or maybe change depends on platform? approx on console but full on PC?

Reply:
It is still a few trascendentals, the approx is very very similar (see the desmos https://www.desmos.com/calculator/5jdfc4pgtk )

Reply:
Check the error diff in the diff, it is very very minimal (bottom of the desmos)